lib/puppet.rb

```diff
@@ -19,6 +19,20 @@
 require 'puppet/external/pson/version'
 require 'puppet/external/pson/pure'
 
+## It would be really nice to use Puppet.features here, but we have to
+## do gettext setup before all of that machinery is loaded. Instead,
+## we will use a simple boolean to handle gettext availability in this
+## file, and drop in a "stub" translation function here.
+begin
+  require 'gettext-setup'
+  GETTEXT_AVAILABLE=true
+rescue LoadError
+  def _(msg)
```

Comment:
`_()` is only one of the functions defined by gettext-setup. Are we intending to duplicate the pluralization function `n_()` too? I don't have something better off the top of my head butI worry about maintainability.